pom.xml

```diff
@@ -54,18 +54,25 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
 
-    <opensaml.version>4.2.0</opensaml.version>
+    <opensaml.version>4.3.0</opensaml.version>
   </properties>
 
   <dependencies>
     <dependency>
       <groupId>javax.servlet</groupId>
       <artifactId>javax.servlet-api</artifactId>
       <version>3.1.0</version>
+      <scope>provided</scope><!-- Must be specified as provided because of the license -->
     </dependency>
+    <dependency>
+      <groupId>jakarta.servlet</groupId>
+      <artifactId>jakarta.servlet-api</artifactId>
+      <version>6.0.0</version>
+      <scope>provided</scope><!-- Must be specified as provided because of the license -->
```

Comment:
I don't see the licensing issue, especially with jakarta, care to elaborate?

Reply:
The Jakarta Servlet API is licensed under EPL 2.0 which is critical and under GPL which is realy strong critical . If you use OSS under GPL license your prooct have to be licensed under GPL as well and this is very bad. That is why I recommend to iclude it just with scope provided.

I would also advise putting this Saml client library under Apache 2.0 licence. The Apache 2.0 licence is the least problematic.
But that's your topic ;-)

Reply:
MIT license is definitely the most permissive though.

IANAL so I've reached out to our legal team for regarding your other comment.

Reply:
@ jebeaudet any movement on this from legal?

Reply:
I got the feedback from our lawyers that including jakarta.servlet-api is fine in the current circumstances. 

I'll get things moving for this PR internally, but I cannot make any promise nor ETA.

Reply:
Thanks for working to get this done. Love the library so glad it will keep up with Java 17 changes!